assets/js/specific/ajaxTextObjectRecSearch/ajax-text-search.js

```diff
@@ -0,0 +1,65 @@
+( function( $ ) {
+
+	$( document ).ready( function() {
+
+		$(".chosen-container input").on('keyup',function( e ){
+
+			if ( !$(this).closest( ".chosen-container" ).prev( "select[data-ajax-txt-search='1']" ).length ) {
+				return;
+			}
+
+			var $selectField = $(this).closest( ".chosen-container" ).prev( "select" );
+			var $inputField = $(this);
+
+			var selectedVal = $selectField.val();
+
+			if ( this.value.length >= 2 && ( ( e.keyCode >= 48 && e.keyCode <= 90 ) || e.keyCode == 8 ) ) {
+
+				var params = {};
+				params[ 'searchTerm'    ] = this.value;
+				params[ 'filterBy'      ] = $selectField.data( 'filter-by' );
+				params[ 'targetObject'  ] = $selectField.data( 'object' );
+				params[ 'dbFilters'     ] = $selectField.data( 'object-filters' );
+				params[ 'orderBy'       ] = $selectField.data( 'order-by' );
+				params[ 'maxRows'       ] = $selectField.data( 'ajax-maxrows' );
+				params[ 'ajaxTxtSearch' ] = $selectField.data( 'ajax-txt-search' );
+
+				// for child select, get parent selected value for filtering
+				if ( typeof params[ 'filterBy' ] != 'undefined' ) {
+					var filterByField = params[ 'filterBy' ];
+
+					var selectedParentVal = $('select[data-filter-child-id*="'+ $selectField.attr( "id" ) +'"]').val();
+
+					params[ filterByField  ] = selectedParentVal;
+				}
+
+				$.ajax({
+					type: 'POST',
+					url : cfrequest.searchTermUrl,
```

Comment:
If there are multiple instances of this form control, they could potentially need/want different search term URLs. Makes sense to put this as a data attribute on the selectField I would suggest.